### PR TITLE
arch: arm: Convert Atmel SAM4S series MCU to use ASF

### PR DESCRIPTION
**Pull request by fallrisk**
Signed-off-by: Justin Watson <jwatson5@gmail.com>

### CODE REVIEW

File: arch/arm/soc/atmel_sam/sam4s/soc.h [hunk 1]
```diff
@@ -13,207 +13,26 @@
...
-#define IRQ_TC3		26	/* Timer Counter Channel #3             */
-#define IRQ_TC4		27	/* Timer Counter Channel #4             */
-#define IRQ_TC5		28	/* Timer Counter Channel #5             */
-#define IRQ_ADC		29	/* ADC Controller                       */
-#define IRQ_DACC	30	/* DAC Controller                       */
-#define IRQ_PWM		31	/* PWM Controller                       */
-#define IRQ_CRCCU	32	/* CRC Controller                       */
-#define IRQ_ACC		33	/*                    */
-#define IRQ_UDP		34	/* USB Device Port                      */
+#define DONT_USE_CMSIS_INIT
```
Comment (galak) [inline arch/arm/soc/atmel_sam/sam4s/soc.h]: This is probably going to need:

#ifndef _ASMLANGUAGE

...
#endif 

The SAME7 code does this in board.h which is wrong and should be cleaned up.

**Review by galak — CHANGES_REQUESTED** [review]
in soc.c we have a mix of __PMC and PMC references  (assume they should all be PMC)

**Review by galak — APPROVED** [review]

File: arch/arm/soc/atmel_sam/sam4s/soc_pinmap.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#include <soc.h>
+
+/* Universal Asynchronous Receiver Transmitter (UART) */
+
+#define PIN_UART0_RXD {PIO_PA9A_UART0_URXD0,  PIOA, ID_PIOA, SOC_GPIO_FUNC_A}
+#define PIN_UART0_TXD {PIO_PA10A_UART0_UTXD0, PIOA, ID_PIOA, SOC_GPIO_FUNC_A}
+
+#define PINS_UART0 {PIN_UART0_RXD, PIN_UART0_TXD}
+
+#define PIN_UART1_RXD {PIO_PB2A_UART1_URXD1,  PIOA, ID_PIOA, SOC_GPIO_FUNC_A}
```
Comment (mnkp) [inline arch/arm/soc/atmel_sam/sam4s/soc_pinmap.h]: This pin is on PIOB, not PIOA. The correct definition is:
```
#define PIN_UART1_RXD {PIO_PB2A_URXD1,  PIOB, ID_PIOB, SOC_GPIO_FUNC_A}
```
The same applies to PIN_UART1_TXD.

**Review by mnkp — COMMENTED** [review]
I'm afraid I was unable to review the code on time. Unfortunately there are several major issues, including the fact that this code will not even compile. This PR should be reverted.

**Pull request merged**